core/processing.go

```diff
@@ -0,0 +1,47 @@
+package core
+
+import (
+	"image"
+	"image/color"
+	"image/draw"
+	"image/png"
+	"os"
+)
+
+func ProduceBlackWhite(inputImg string) {
+	file, err := os.Open(inputImg)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	img, _, err := image.Decode(file)
+	if err != nil {
+		panic(err)
+	}
+
+	newImg := image.NewRGBA(img.Bounds())
+	draw.Draw(newImg, newImg.Bounds(), img, image.Point{}, draw.Src)
+
+	for x := range img.Bounds().Dx() + 2 {
+		for y := range img.Bounds().Dy() + 2 {
+			oldPixel := img.At(x, y)
+			r, g, b, _ := oldPixel.RGBA()
+			lum := 0.299*float64(r) + 0.587*float64(g) + 0.114*float64(b)
```

Comment:
How did you determine these constants?

Reply:
So the most basic method to convert to grayscale is just dividing the sum of the RGB values by 3. There's a whole bunch of formulas like the above that are improved versions which tweak the individual weights and have slightly different results depending on what you're looking for. Just grabbed one of them like here: https://stackoverflow.com/questions/596216/formula-to-determine-perceived-brightness-of-rgb-color